components/script/indexed_db.rs

```diff
@@ -35,9 +38,16 @@ pub fn key_type_to_jsval(
         IndexedDBKeyType::Number(n) => result.set(DoubleValue(*n)),
         IndexedDBKeyType::String(s) => s.safe_to_jsval(cx, result),
         IndexedDBKeyType::Binary(b) => b.safe_to_jsval(cx, result),
-        IndexedDBKeyType::Date(_d) => {
-            // TODO: implement this when Date's representation is finalized.
-            result.set(UndefinedValue());
+        IndexedDBKeyType::Date(d) => unsafe {
+            // TODO: This doesn't clip the time (while firefox does)
+            // Inner representation of clippedtime is a f64, this is safe
+            let time: ClippedTime = std::mem::transmute(d.trunc());
+            // Could be null
+            let date = js::jsapi::NewDateObject(*cx, time);
```

Comment:
Please add a CanGc argument to this function, since we can allocate new JS objects.